### ISSUE TITLE
Youki as runtime 

### ISSUE THREAD

**Issue by Ludea** [fussybeaver/bollard#295]
It would be awesome if you could support [Youki](https://github.com/containers/youki) as container runtime

**Comment by fussybeaver** [comment]
That is really interesting I hadn't heard of that project. Initially not quite sure how or if an integration would work - e.g. can you use Youki as a library without Docker/Portman ? I'll need to look a little closer..

**Comment by Ludea** [comment]
You can run youki as standalone app : https://containers.github.io/youki/user/basic_usage.html#using-youki-standalone

It can create OCI container https://github.com/containers/youki/tree/main/crates/liboci-cli.

You can also replace docker runtime (runc) , with youki.

I think, the best way to integrate youki to bollard is to provide a cli to parse dockerfile and create image through youki with dockerfile.

**Comment by fussybeaver** [comment]
Parsing the Dockerfile is big enough to be a project on its own (e.g. buildkit)... I can imagine perhaps translating the API and struct definitions in Bollard into Youki configuration - although you are missing the rootfs file structure to actually run it. I guess the challenge is finding out what it takes to do that.

**Comment by Ludea** [comment]
We can get remote rootfs with container config 
https://containers.gitbook.io/build-containers-the-hard-way/#container-configuration

**Comment by ledoyen** [comment]
Hello :wave:, isn't youki (or any other alternative runtime) already supported through the use of the `HostConfig#runtime` field, when creating a new container ?

**Comment by fussybeaver** [comment]
@ledoyen I think you're right, if you setup your docker daemon to use youki as a runtime. Perhaps though it might be more useful to be able to trigger youki directly from Bollard, as you will not need the Docker server at all.

**Comment by Ludea** [comment]
> @ledoyen I think you're right, if you setup your docker daemon to use youki as a runtime. Perhaps though it might be more useful to be able to trigger youki directly from Bollard, as you will not need the Docker server at all.

Yes, it would be awesome to avoid Docker and to get a native rust container engine, with cli support like buildkit
